tests/integration/mixer/policy/ratelimiting_test.go

```diff
@@ -59,171 +61,218 @@ func TestRateLimiting_RedisQuotaRollingWindow(t *testing.T) {
-			if got429s < want429s {
-				attributes := []string{fmt.Sprintf("%s=\"%s\"", util.GetDestinationLabel(),
-					util.Fqdn(destinationService, bookInfoNameSpaceStr)),
-					fmt.Sprintf("%s=\"%d\"", util.GetResponseCodeLabel(), 429),
-					fmt.Sprintf("%s=\"%s\"", util.GetReporterCodeLabel(), "destination")}
-				t.Logf("prometheus values for istio_requests_total for 429's:\n%s",
-					util.PromDumpWithAttributes(prom, "istio_requests_total", attributes))
-				t.Errorf("Bad metric value for rate-limited requests (429s): got %f, want at least %f", got429s,
-					want429s)
+			err := validateDefaultLessThanOverride(res, t, destinationService, bookInfoNameSpaceStr)
+			if err != nil {
+				t.Logf("Got error, trying again. Err: %v", err)
+				res = util.SendTraffic(ing, t, "Sending traffic...", "", "", 300)
+				err = validateDefaultLessThanOverride(res, t, destinationService, bookInfoNameSpaceStr)
+				if err != nil {
+					// Fail now, as its still error'ing out.
+					t.Fatalf("Failed with err %v", err)
+				}
 			}
 		})
 }
 
 func testRedisQuota(t *testing.T, config bookinfo.ConfigFile, destinationService string) {
-	framework.NewTest(t).Label(label.Flaky).Run(func(ctx framework.TestContext) {
+	framework.Run(t, func(ctx framework.TestContext) {
 		g.ApplyConfigOrFail(
 			t,
 			bookinfoNs,
 			bookinfo.NetworkingReviewsV3Rule.LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
 		)
-		defer g.DeleteConfigOrFail(t,
-			bookinfoNs,
-			bookinfo.NetworkingReviewsV3Rule.LoadWithNamespaceOrFail(t, bookinfoNs.Name()))
 		bookInfoNameSpaceStr := bookinfoNs.Name()
 		config := setupConfigOrFail(t, config, bookInfoNameSpaceStr, red, g, ctx)
-		defer deleteConfigOrFail(t, config, g, ctx)
+		defer func() {
+			g.DeleteConfigOrFail(t,
+				bookinfoNs,
+				bookinfo.NetworkingReviewsV3Rule.LoadWithNamespaceOrFail(t, bookinfoNs.Name()))
+			deleteConfigOrFail(t, config, g, ctx)
+			util.AllowRuleSync(t)
+		}()
 		util.AllowRuleSync(t)
 
-		// This is the number of requests we allow to be missing to be reported, so as to make test stable.
-		errorInRequestReportingAllowed := 5.0
-		_ = util.SendTraffic(ing, t, "Sending traffic...", "", "", 300)
-		prior429s, prior200s := util.FetchRequestCount(t, prom, destinationService, "",
-			bookInfoNameSpaceStr, 300)
+		errorInRequestReportingAllowed = 5
+		util.ValidateBookInfoGatewayIsSetup(t, prom, bookinfoNs)
+		res := util.SendTrafficAndWaitForExpectedStatus(ing, t, "Sending traffic...", "", 300, http.StatusOK)
+		util.AllowPrometheusSync(t)
+		err := validate200sand429s(res, t, 0, destinationService, bookInfoNameSpaceStr)
+		if err != nil {
+			t.Logf("Got error, trying again. Err: %v", err)
+			// Try again as prometheus can take time to set up.
+			res = util.SendTrafficAndWaitForExpectedStatus(ing, t, "Sending traffic...", "", 300, http.StatusOK)
+			util.AllowPrometheusSync(t)
+			err = validate200sand429s(res, t, prior429s+prior200s+300-errorInRequestReportingAllowed, destinationService, bookInfoNameSpaceStr)
```

Comment:
can we move the `prior429s+prior200s+300-errorInRequestReportingAllowed` to a variable just to make this easier to parse?